### PR TITLE
Improve efficiency of (de)marshalling byte[] #405

### PR DESCRIPTION
**Pull request by peter-lawrey**

### CODE REVIEW

File: src/main/java/net/openhft/chronicle/wire/BinaryWire.java [hunk 1]
```diff
@@ -2699,7 +2699,7 @@ public WireIn bytes(@NotNull ReadBytesMarshallable bytesConsumer) {
 
         @NotNull
         @Override
-        public byte @NotNull [] bytes() {
+        public byte @NotNull [] bytes(byte[] using) {
```
Comment (RobAustin) [inline src/main/java/net/openhft/chronicle/wire/BinaryWire.java:2702]: found `byte[] using` be @NotNull as well

Comment (RobAustin) [reply src/main/java/net/openhft/chronicle/wire/BinaryWire.java:2702]: is it a good idea to make the method have a `using` that takes an array[]. as they are a fixed size ( maybe smaller than you require ), maybe the method should not be public but packaged local.

**Review by RobAustin — APPROVED** [review]

File: src/main/java/net/openhft/chronicle/wire/BinaryWire.java (hunk 1 above)
Comment (peter-lawrey) [reply src/main/java/net/openhft/chronicle/wire/BinaryWire.java:2702]: This `using` only works if the size is the same. 
In some benchmarks at least, this is the case 100% of the time.

Comment (peter-lawrey) [reply src/main/java/net/openhft/chronicle/wire/BinaryWire.java:2702]: The better solution is to use `Bytes`

**Pull request merged**